product/views.py

```diff
@@ -0,0 +1,182 @@
+            products = Product.objects.filter(goal=Goal.objects.get(name__icontains=sort))
+            result   = self.make_product_info_list(products=products) 
+            
+            return JsonResponse({"data": result, "message": "SUCCESS"}, status=200)
+
+        if sort == 'new':
+            products = Product.objects.filter(is_new=bool(sort == 'new'))
+            result   = self.make_product_info_list(products=products) 
+
+            return JsonResponse({"data": result, "message": "SUCCESS"}, status=200)
+
+        return JsonResponse({"message": "PAGE_NOT_FOUND"}, status=404)
+
+        
+class ProductDetailView(View):
+    def get(self, request, product_id):
+        if not Product.objects.filter(id=product_id).exists():
+            return JsonResponse({'message': 'DOES_NOT_EXIST'}, status=404)
+
+        product  = Product.objects.get(id=product_id)
+        category_name = product.category.menu.name
+        context  = {
+                'category'           : category_name,
+                'id'                 : product.id,
+                'productImageSrc'    : product.image_set.get(is_main=False).image_url,
+                'productCardImageSrc': product.image_set.get(is_main=True).image_url,
+                'healthGoalList'     : [goal.name for goal in product.goal.all()],
+                'title'              : product.name,
+                'subTitle'           : product.sub_name,
+                'description'        : product.description,
+                'nutritionLink'      : product.nutrition_url,
+                'allergyList'        : [allergy.name for allergy in product.allergy.all()],
+                'dietaryHabitList'   : [dietary_habit.name for dietary_habit in product.dietary_habit.all()],
+                }
+
+        # is_vegan, is_vegeterian
+        vegan_level = product.vegan_level_id
+        if vegan_level == 1:
+            is_vegan      = True
+            is_vegeterian = False
+        elif vegan_level == 2:
+            is_vegan      = False
+            is_vegeterian = True
+        else:
+            is_vegan      = False
+            is_vegeterian = False
+
+        context['isVegan']        = is_vegan
+        context['isVegetarian']   = is_vegeterian
+
+        # size, price
+        product_SSPs = ProductStock.objects.filter(product=product)  # SSP: siz e, stock, price
+        if category_name == 'vitamins':
+            price      = product_SSPs.first().price   
+            is_soldout = bool(product_SSPs.first().stock == 0)
+        else:
+            price      = {product_SSP.size : product_SSP.price for product_SSP in product_SSPs}
+            is_soldout = {product_SSP.size : bool(product_SSP.stock == 0) for product_SSP in product_SSPs}
+
+        context['productPrice']     = price
+        context['isSoldOut']        = is_soldout
+
+        # similar products
+        goals_product        = product.goal.all()
+        for goal_product in goals_product:
```

Comment:
```suggestion
        for goal_product in product.goal.all():
```
goals_product 해당 변수가 여러번 사용 되지 않기 때문에 변수를 선언하지 않아도 될 것으로 보입니다.